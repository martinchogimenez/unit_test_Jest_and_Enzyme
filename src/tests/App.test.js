import React from "react";
import { shallow } from "enzyme";
import App from "../App";



//Test suite Cartas obtener usuarios
describe('Test suite Cartas obtener usuarios', function () {

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<App />);
    })

    // Test case a- App debe coincidir con snapshot
    test('App debe coincidir con snapshot', function () {

        expect(wrapper).toMatchSnapshot();
        
    });

   
    //Test case b- el título debe ser "Listado de Usuarios" 
    test('el título debe ser "Listado de Usuarios"', function () {
           
        expect(wrapper.find("h2").text()).toEqual("Listado de Usuarios");

    });

    // Test case c- debe existir un loader con texto "cargando página..."
    test('debe existir un loader con texto "cargando página..."', function () {
        
        expect(wrapper.contains(<p>Cargando página...</p>)).toBe(true);

    });


    //Test case h- el color de fondo debe ser inicialmente "#ccffff"
    test('el color de fondo debe ser inicialmente "#ccffff"', function () {
          
        expect(wrapper.state().color).toEqual("#ccffff");

    });

    //Test case i- el color de fondo debe ser "#ffff99" al hacer click en el div contenedor
    test('el color de fondo debe ser "#ffff99" al hacer click en el div contenedor', function () {
        
        wrapper.find("div").first().simulate("click");

        expect(wrapper.state().color).toEqual("#ffff99");

    });
     
});