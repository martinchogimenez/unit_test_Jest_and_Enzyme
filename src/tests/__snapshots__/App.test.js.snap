// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite Cartas obtener usuarios App debe coincidir con snapshot 1`] = `ShallowWrapper {}`;
